core/amqp_leader-1.0.0/src/amqp_leader_listener.erl

```diff
@@ -0,0 +1,241 @@
+%%                                   {stop, Reason, Reply, State} |
+%%                                   {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_call(any(), {pid(), any()}, state()) -> handle_call_ret_state(state()).
+handle_call('is_ready', From, #state{pending = Pids} = State) ->
+    NewState = maybe_ready(State#state{pending = [From | Pids]}),
+    {'noreply', NewState};
+handle_call(_Request, _From, State) ->
+    lager:warning("unhandled call ~p", [_Request]),
+    {'reply', {'error', 'not_implemented'}, State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling cast messages
+%%
+%% @spec handle_cast(Msg, State) -> {noreply, State} |
+%%                                  {noreply, State, Timeout} |
+%%                                  {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_cast(any(), state()) -> handle_cast_ret_state(state()).
+handle_cast({'is_ready', Pid, Ref}, #state{pending = Pending} = State) ->
+    NewState = maybe_ready(State#state{pending = [{Pid, Ref} | Pending]}),
+    {'noreply', NewState};
+handle_cast({'wh_nodes', {'expire', #wh_node{node = Node}}}, #state{name = Name} = State) ->
+    Name ! {'DOWN', Node},
+    {'noreply', State};
+handle_cast({'gen_listener', {'created_queue', _QueueNAme}}, State) ->
+    lager:debug("connected"),
+    NewState = maybe_ready(State#state{has_queue = 'true'}),
+    {'noreply', NewState};
+handle_cast({'gen_listener', {'is_consuming', _IsConsuming}}, State) ->
+    lager:debug("consuming"),
+    NewState = maybe_ready(State#state{is_consuming = 'true'}),
+    {'noreply', NewState};
+handle_cast(_Msg, State) ->
+    lager:warning("unhandled cast ~p", [_Msg]),
+    {'noreply', State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling all non call/cast messages
+%%
+%% @spec handle_info(Info, State) -> {noreply, State} |
+%%                                   {noreply, State, Timeout} |
+%%                                   {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_info(any(), state()) -> handle_info_ret_state(state()).
+handle_info(_Info, State) ->
+    lager:warning("~s unhandled info ~p", [node(), _Info]),
+    {'noreply', State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Allows listener to pass options to handlers
+%%
+%% @spec handle_event(JObj, State) -> {reply, Options}
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_event(wh_json:object(), state()) -> {'reply', []}.
+handle_event(JObj, #state{name = Name}) ->
+    wh_util:put_callid(wapi_leader:queue(Name, node())),
+    NodeBin = iolist_to_binary(io_lib:format("~s", [node()])),
```

Comment:
@siplabs
To get the node as a binary you can utilize the wh_util functions ;)
NodeBin = wh_util:to_binary(node()),